src/Conversion/ONNXToTorch/NN/ReduceMean.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------- ReduceMean.cpp - ONNX Op Transform ------------------===//
+//
+// Copyright 2019-2020 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file implements a combined pass that dynamically invoke several
+// transformation on ONNX ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTorch/NN/CommonUtils.h"
+#include "src/Conversion/ONNXToTorch/ONNXToTorchCommon.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+#include <fstream>
+#include <iostream>
+#include <set>
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Interfaces/CallInterfaces.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Support/FileUtilities.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/Support/MD5.h"
+#include "llvm/Support/ToolOutputFile.h"
+
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/Interface/ShapeInferenceOpInterface.hpp"
+#include "src/Pass/Passes.hpp"
+#include "src/Support/OMOptions.hpp"
+
+#include "mlir/Transforms/DialectConversion.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchDialect.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/Transforms/Passes.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+#include "llvm/ADT/StringExtras.h"
+
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionDialect.h"
+#include "torch-mlir/Dialect/TorchConversion/IR/TorchConversionOps.h"
+#include "torch-mlir/Dialect/TorchConversion/Transforms/BackendTypeConversion.h"
+
+#ifdef _WIN32
+#include <io.h>
+#endif
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
```

Comment:
@ramakota-helprack 
This file needs a clang-format, especially for the order of inclusion of the headers.

Also, a comment on the lines of what is there in the older files is needed here.  This should typically be a copy-paste of the documentation of the ONNX operation on the web.

Reply:
Resolved